src/main/java/org/truffleruby/core/rubinius/StatNodes.java

```diff
@@ -26,6 +30,15 @@
 @CoreClass("File::Stat")
 public abstract class StatNodes {
 
+    @TruffleBoundary
+    private static String getPathStringHandlingBinary(DynamicObject path) {
+        if (StringOperations.encoding(path) == ASCIIEncoding.INSTANCE) {
+            return RopeOperations.decodeRope(StandardCharsets.UTF_8, StringOperations.rope(path));
```

Comment:
Doesn't this put us back into a similar situation as the removed `decodeUTF8` method?

Reply:
Yes, but only for these 2 usages and only if the String has a binary encoding.

Reply:
It will go away when stat() is moved to NFI, naturally.